docs/content/guides/dagster-pipes/modifying-existing-code-to-work-with-dagster-pipes.mdx

```diff
@@ -1,6 +1,270 @@
+
+</Note>
+
+## Prerequisites
+
+<Note>
+
+This guide assumes your code is being orchestrated by Dagster. If that’s not the case, please refer to <a href="integrating-subprocess-with-dagster-pipes">Integrating Dagster Pipes with Subprocess</a> for how to orchestrate your existing Python scripts using Dagster.
+
+</Note>
+
+To get started, you’ll need:
+
+- To install the dagster-pipes library:
+
+```bash
+pipe install dagster-pipes
+```
+
+- **An existing Python script.** We’ll use the following Python script to demonstrate:
+
+```python file=/guides/dagster/dagster_pipes/subprocess/external_code_original.py
+# ruff: noqa: T201
+
+import pandas as pd
+
+
+def main():
+    orders_df = pd.DataFrame({"order_id": [1, 2], "item_id": [432, 878]})
+    total_orders = len(orders_df)
+    print(f"processing total {total_orders} orders")
+
+
+if __name__ == "__main__":
+    main()
+```
+
+---
+
+## About Dagster
+
+### Materialize Dagster assets
+
+In Dagster, data assets - like a database table - are created during a process called _materialization._ During materialization, Dagster runs the function in the asset’s body to create the asset. In this case, the ‘function’ is your existing code.
+
+Dagster Pipes automatically makes some pieces of data about assets available, such as those about materialization or data quality checks. You can also send your own structured metadata and execution logs back to Dagster. We’ll come back to this in later sections.
+
+When your existing code is orchestrated by Dagster, you can run it from the Dagster UI. Navigate to the the corresponding asset page and click **Materialize** to run your code. Refer to [Viewing and materializing assets in the UI](/concepts/assets/software-defined-assets#viewing-and-materializing-assets-in-the-ui) for more info.
+
+<Image
+alt="Materialize Dagster assets"
+src="/images/guides/dagster-pipes/modifying-code-1-materialize-dagster-assets.png"
+width={1328}
+height={836}
+/>
+
+### Monitor code execution in Dagster UI
+
+To monitor the run, navigate to the [Run details page](/concepts/webserver/ui#run-details). This page includes timing, raw compute logs, and structured events. The bottom section shows filterable logs from the run, detailing events, their types, and specifics. There are two types of log: structured event and raw compute.
+
+#### Structured event logs
+
+Structured logs are enriched and categorized with metadata. For example, a label of what the log entry is about, links to an asset’s metadata, and the event type. This detailed structuring not only facilitates easier filtering and searching within the logs, but also populates the asset catalog and status information in the asset graph. We’ll come back to how you can log metadata from your existing code to Dagster in a little bit.
```

Comment:
@erinkcochran87 and i landed on this because we thought the audience (stakeholders) to this page may not be familiar with dagster, so the writing here is intentionally to be a subset of generalized dagster docs